packages/amplify-e2e-core/src/categories/auth.ts

```diff
@@ -1340,6 +1340,63 @@ export function updateAuthAddUserGroups(projectDir: string, groupNames: string[]
   });
 }
 
+export function updateAuthAddUserGroupsAfterPull(
+  projectDir: string,
+  groupNames: string[],
+  settings?: {
+    testingWithLatestCodebase: boolean;
+  },
+): Promise<void> {
+  if (groupNames.length == 0) {
+    return undefined;
+  }
+  const testingWithLatestCodebase = settings?.testingWithLatestCodebase || false;
+  return new Promise((resolve, reject) => {
+    const chain = spawn(getCLIPath(testingWithLatestCodebase), ['update', 'auth'], { cwd: projectDir, stripColors: true });
+
+    chain
+      .wait('What do you want to do?')
+      .send(KEY_DOWN_ARROW)
+      .send(KEY_DOWN_ARROW)
+      .send(KEY_DOWN_ARROW)
+      .sendCarriageReturn()
+      .wait('Select any user pool groups you want to delete:')
+      .sendCarriageReturn()
+      .wait('Do you want to add another User Pool Group')
+      .sendConfirmYes()
+      .wait('Provide a name for your user pool group')
+      .send(groupNames[0])
+      .sendCarriageReturn();
+
+    if (groupNames.length > 1) {
+      let index = 1;
+      while (index < groupNames.length) {
+        chain
+          .wait('Do you want to add another User Pool Group')
+          .sendConfirmYes()
+          .wait('Provide a name for your user pool group')
+          .send(groupNames[index++])
+          .sendCarriageReturn();
+      }
+    }
+
+    chain
+      .wait('Do you want to add another User Pool Group')
+      .sendCarriageReturn()
+      .wait('Sort the user pool groups in order of preference')
+      .sendCarriageReturn()
+      .wait('"amplify publish" will build all your local backend and frontend resources');
+
+    chain.run((err: Error) => {
+      if (!err) {
+        resolve();
+      } else {
+        reject(err);
+      }
+    });
```

Comment:
nit: This can simplify to `chain.runAsync();`

With that change, the function can become an async function instead of returning a `new Promise(...)`.